service/src/main/java/bio/terra/cbas/runsets/inputs/InputGenerator.java

```diff
@@ -51,12 +52,15 @@ public static Map<String, Object> buildInputs(
                 attributeName, recordResponse.getId(), parameterName);
           }
         }
+      } else {
+        throw new WorkflowInputSourceNotSupportedException(param.getSource());
       }
 
-      // Convert into an appropriate CbasValue:
-      CbasValue cbasValue = CbasValue.parseValue(param.getInputType(), parameterValue);
-
-      params.put(parameterName, cbasValue.asSerializableValue());
+      if (parameterValue != null) {
+        // Convert into an appropriate CbasValue:
+        CbasValue cbasValue = CbasValue.parseValue(param.getInputType(), parameterValue);
+        params.put(parameterName, cbasValue.asSerializableValue());
+      }
```

Comment:
For reviewers: these are the lines that actually implement the requested feature update